### PR TITLE
10.2.1 Release Notes

### PR DESCRIPTION
**Pull request by pmaier1**
Draft for 10.2.1 Release notes

### CODE REVIEW

**Comment by settermjd** [comment]
@pmaier1, I squashed your 3 commits into one to keep the commit history as compact as possible. @patrickjahns @phil-davis, can you review, time-permitting, pls?

**Comment by micbar** [comment]
@pmaier1 We will include https://github.com/owncloud/core/pull/35595 in the release.

Original Ticket in https://github.com/owncloud/enterprise/issues/3316

**Comment by settermjd** [comment]
> @pmaier1 We will include [owncloud/core#35595](https://github.com/owncloud/core/pull/35595) in the release.
> 
> Original Ticket in [owncloud/enterprise#3316](https://github.com/owncloud/enterprise/issues/3316)

@micbar, I've just added a reference to that issue as fixed in the release notes; see 843b7db.

File: modules/admin_manual/pages/release_notes.adoc [hunk 1]
```diff
@@ -44,16 +44,22 @@ Version 10.2.1 corrects this behavior by dropping `share keys` for master key en
...
 Version 10.2.0 {release-notes-known-issues-url}[accidentally changed the location of user avatars] making them unavailable and storing uploaded avatar images in the wrong location.
 10.2.1 restores the earlier behavior and provides a repair step to move back the avatar images uploaded with 10.2.0 to the right location. 
 As it is not necessary nor possible to run `occ upgrade` when upgrading from 10.2.0 to this patch release, **you need to run `occ maintenance:repair -s 'OC\Repair\MoveAvatarIntoSubFolder'` manually to trigger the repair step**.
 * **Fixed https://doc.owncloud.com/server/admin_manual/release_notes.html#known-issues[known issue] with "Password changed" HTML emails rendered in plain text**
 * **Fixed use of invalid token on password reset** +
 Password reset links sent to a user were invalid, if the user attempted to login using their e-mail address and an invalid password prior to filling out and submitting the Reset Password form.
+* **Fixed issue when removing a user from a group**
+Remowing a user from a group using the user management UI resulted in an error that prevented the page to be refreshed to show the changes. This has been taken care of.
```
Comment (settermjd) [inline modules/admin_manual/pages/release_notes.adoc]: This sentence doesn't quite read correctly. I suggest changing it to:: Remowing a user from a group using the user management UI resulted in an error that prevented the page to be refreshed from showing the changes. This has been corrected.

Comment (phil-davis) [reply modules/admin_manual/pages/release_notes.adoc]: IMO the problem was that the changes did not show in real-time, and you had to refresh the page to see that the action really worked?
If that is so, then:

Removing a user from a group using the user management UI resulted in an error that required the page to be refreshed to show the changes. This has been corrected.

File: modules/admin_manual/pages/release_notes.adoc [hunk 2]
```diff
@@ -21,6 +24,45 @@
...
+Previously both modes used the same mechanisms which resulted in potentially significant overhead when master key encryption was used as user key encryption relies on so-called `share keys` which are necessary to allow share recipients to decrypt shared files.
+
+With master key encryption, `share keys` are redundant as you have one central key that can be used to decrypt all files. 
+Version 10.2.1 corrects this behavior by dropping `share keys` for master key encryption, thereby increasing the performance dramatically, especially when sharing folders with many files as said keys do not have to be generated anymore for each file.
+
+=== Solved Issues
+
+* **Fixed reshare permission issue** +
+ownCloud Server 10.2.0 introduced an issue that allows users to increase sharing permissions beyond their own permissions in a reshare scenario: When user A shares a folder "_Project_" with user B, granting only read and share permission, then the Sharing API allowed user B to reshare a subfolder of "_Project_" with user C granting full permissions or to create a public link on the shared folder, respectively.
```
Comment (micbar) [inline modules/admin_manual/pages/release_notes.adoc]: @settermjd @pmaier1 This has not been introduced in 10.2.0. It has been undiscovered for a long time.

File: modules/admin_manual/pages/release_notes.adoc (hunk 1 above)
Comment (pmaier1) [reply modules/admin_manual/pages/release_notes.adoc]: yes, thx both.

File: modules/admin_manual/pages/release_notes.adoc (hunk 2 above)
Comment (pmaier1) [reply modules/admin_manual/pages/release_notes.adoc]: Thx, changed.

**Comment by pmaier1** [comment]
Thanks! Feedback has been addressed. Please check again if everything is included and backport to all oC 10 versions together with the release. :v:

File: modules/admin_manual/pages/release_notes.adoc [hunk 3]
```diff
@@ -21,6 +24,45 @@
...
+10.2.1 restores the earlier behavior and provides a repair step to move back the avatar images uploaded with 10.2.0 to the right location. 
+As it is not necessary nor possible to run `occ upgrade` when upgrading from 10.2.0 to this patch release, **you need to run `occ maintenance:repair -s 'OC\Repair\MoveAvatarIntoSubFolder'` manually to trigger the repair step**.
+* **Fixed https://doc.owncloud.com/server/admin_manual/release_notes.html#known-issues[known issue] with "Password changed" HTML emails rendered in plain text**
+* **Fixed use of invalid token on password reset** +
+Password reset links sent to a user were invalid, if the user attempted to login using their e-mail address and an invalid password prior to filling out and submitting the Reset Password form.
+* **Fixed issue when removing a user from a group** +
+Removing a user from a group using the user management UI resulted in an error that required the page to be refreshed to show the changes. This has been corrected.
+* **Added -y option to `occ encryption:encrypt-all` command +
```
Comment (phil-davis) [inline modules/admin_manual/pages/release_notes.adoc]:
```suggestion
* **Added -y option to `occ encryption:encrypt-all` command** +
```

File: modules/admin_manual/pages/release_notes.adoc [hunk 4]
```diff
@@ -21,6 +24,45 @@
...
+This undesired behavior is fixed with 10.2.1.
+* **Fixed issue with Sharing API and enforced public link expiration dates** +
+An issue in the Sharing API caused the ownCloud clients to prevent users from creating public links when the option "Enforce expiration date" for public links is in use. This is now solved.
+* **Fixed known issue with user avatar paths** +
+Version 10.2.0 {release-notes-known-issues-url}[accidentally changed the location of user avatars] making them unavailable and storing uploaded avatar images in the wrong location.
+10.2.1 restores the earlier behavior and provides a repair step to move back the avatar images uploaded with 10.2.0 to the right location. 
+As it is not necessary nor possible to run `occ upgrade` when upgrading from 10.2.0 to this patch release, **you need to run `occ maintenance:repair -s 'OC\Repair\MoveAvatarIntoSubFolder'` manually to trigger the repair step**.
```
Comment (phil-davis) [inline modules/admin_manual/pages/release_notes.adoc]:
```suggestion
As it is not necessary nor possible to run `occ upgrade` when upgrading from 10.2.0 to this patch release, if you are already running 10.2.0 then after installing 10.2.1 **you need to run `occ maintenance:repair -s 'OC\Repair\MoveAvatarIntoSubFolder'` manually to trigger the repair step**.
```
Maybe could clarify, like my suggestion above, that the repair command is only needed if you are already running 10.2.0 and then installing 10.2.1.

For people who upgrade directly from 10.1.* to 10.2.1, they will never have code with the avatar bug, so there should be nothing to repair.

**Comment by phil-davis** [comment]
IMO the 9 commits here can now be squashed. (Let me know if you want me to do that)

**Comment by patrickjahns** [comment]
@settermjd @pmaier1 - with 10.2.1RC3 another addition to the changelog was made - https://github.com/owncloud/core/commit/dc3d09c2f3adeab20d6c975d7301687c349afa23

**Pull request merged**